Sources/AblyAssetTrackingPublisher/PublisherTrackableState.swift

```diff
@@ -0,0 +1,44 @@
+ //
+//  Created by Łukasz Szyszkowski on 25/08/2021.
+//
+
+import Foundation
+
+struct PublisherTrackableState {
```

Comment:
I have a programming question: When should you use classes vs mutable structs? aka. Why did you use `struct` instead of class?

Reply:
`struct` is OK here, it is lightweight and it's perfect for data "storage". It's a value type and their performance is better than `class` which is a reference type. 
Value types (`structs`) are stored on the `Stack` but the reference types (`class`) are stored on `Heap`, `Stack` is faster than `Heap`
The "problem" with structs is that they are [copied on write](https://medium.com/@nitingeorge_39047/copy-on-write-in-swift-b44949436e4f) - as I mentioned above, it's ok for use as a state.
more [HERE](https://abhimuralidharan.medium.com/difference-between-value-type-and-a-reference-type-in-ios-swift-18cb5145ad7a)
I know there is a `class` type state in my code, the only reason why I used them is for testing purposes. I injected them to the state manager and I can check their actual state (eg collection capacity) by reference. This is something that is hard to check when you are using `struct` which is copied on write. To be consistent I have to change all `class` types to `struct` types and change tests a little bit.

Reply:
All methods are mutating the struct, causing a new struct to be created. I don't think its fair to call it lightweight, or better performance at this point.